core/server/master/src/main/java/alluxio/master/file/meta/Inode.java

```diff
@@ -325,6 +335,18 @@ public T setMode(short mode) {
     return getThis();
   }
 
+  /**
+   * Sets the UFS last modification time to the new time if the new time is more recent.
+   * This method can be called concurrently with deterministic results.
+   *
+   * @param ufsLastModificationTimeMs the ufs last modification time to use
+   * @return the updated object
+   */
+  public T setUfsLastModificationTimeMs(long ufsLastModificationTimeMs) {
+    mUfsLastModificationTimeMs = ufsLastModificationTimeMs;
```

Comment:
do we need to check whether the new time is more recent?

Reply:
I updated it.